dev/AppNotifications/AppNotificationTelemetry.h

```diff
@@ -0,0 +1,319 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <WindowsAppRuntimeInsights.h>
+#include <wrl\wrappers\corewrappers.h>
+
+DECLARE_TRACELOGGING_CLASS(AppNotificationTelemetryProvider,
+    "Microsoft.WindowsAppSDK.Notifications.AppNotificationTelemetry",
+    // {7c1b07ef-a7c0-56d1-5456-385ebd4412b2}
+    (0x7c1b07ef, 0xa7c0, 0x56d1, 0x54, 0x56, 0x38, 0x5e, 0xbd, 0x44, 0x12, 0xb2)); //ELx need a new guid here
+
+
+class AppNotificationTelemetry : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TELEMETRY_CLASS(AppNotificationTelemetry, AppNotificationTelemetryProvider);
+
+public:
+    DEFINE_EVENT_METHOD(RegisterByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "RegisterByAPI",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
```

Comment:
I would imagine `AppId` is sufficient. What's the reason of logging `AppName` as well?

Reply:
Or maybe AppName is the DisplayName for unpackaged apps?

Reply:
If you look at the table in the description, you'll see that for packaged app appId and appName are the same but for unpackaged, they are **TestAppId** and **ToastNotificationsTestApp.exe** respectively.

Reply:
PushNotification telemetry is only logging the appName. That's something we'll likely have to revisit later when we go through the comprehensive notification tasks.